statechart/models/statechart_mixin.py

```diff
@@ -316,7 +316,8 @@ class can override the statechart with another one adding new events,
 
     @api.model
     def _get_sc_event_allowed_field_names(self):
-        event_names = self._statechart.events_for()
+        ignore_for_has_allowed_events = self.env.context.get("ignore_for_has_allowed_events", [])
+        event_names = [name for name in self._statechart.events_for() if name not in ignore_for_has_allowed_events]
```

Comment:
nit: you can put the `if name not in ignore_for_has_allowed_events` in the `return` statement, to avoid creating a temporary list.